Backport of upstream bugfix PR#3039 [0] for compatibility with newer
versions of ncurses.

[0] https://github.com/ranger/ranger/pull/3039/files

--- a/ranger/gui/ui.py
+++ b/ranger/gui/ui.py
@@ -151,10 +151,6 @@ def suspend(self):
         self.win.keypad(0)
         curses.nocbreak()
         curses.echo()
-        try:
-            curses.curs_set(1)
-        except curses.error:
-            pass
         if self.settings.mouse_enabled:
             _setup_mouse({"value": False})
         curses.endwin()

--- a/ranger/gui/ui.py
+++ b/ranger/gui/ui.py
@@ -128,9 +128,10 @@ def initialize(self):
             self.is_set_up = True
             self.setup()
             self.win.addstr("loading...")
-            self.win.refresh()
             self._draw_title = curses.tigetflag('hs')  # has_status_line
 
+        self.win.refresh()
+
         self.update_size()
         self.is_on = True
 
